src/twisted/conch/ssh/keys.py

```diff
@@ -1262,17 +1306,27 @@ def blob(self):
             )
         elif type == "EC":
             byteLength = (self._keyObject.curve.key_size + 7) // 8
-            return (
-                common.NS(data["curve"])
-                + common.NS(data["curve"][-8:])
+            curve = data["curve"][-8:]
+            if self._sk:
+                # Format example "sk-ecdsa-sha2-nistp256@openssh.com" -> "nistp256"
+                curve = data["curve"][-20:-12]
```

Comment:
I don't understand this comment. sorry.

Let me know if my comment make sense.
We use the same "extraction" method for all key types.

```suggestion
            # We convert the curve name. 
            # The curve name is only `nistpNNN` part.
            # Format example:
            # "ecdsa-sha2-nistp256" -> "nistp256"
            # "sk-ecdsa-sha2-nistp256@openssh.com" -> "nistp256"
            curve = data["curve"][-8:]
            if self._sk:
                curve = data["curve"][-20:-12]
```

Reply:
Yes, your comment makes more sense. I just wanted to explain the numbers -20 and -12